src/main/java/dev/j3fftw/headlimiter/HeadLimiter.java

```diff
@@ -12,12 +16,19 @@
 import org.bukkit.block.Block;
 import org.bukkit.block.BlockState;
 import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
 import org.bukkit.event.block.BlockPlaceEvent;
 import org.bukkit.plugin.java.JavaPlugin;
 
 public final class HeadLimiter extends JavaPlugin implements Listener {
 
+    ThreadFactory threadFactory = new ThreadFactoryBuilder().setNameFormat("HeadLimiter-thread").build();
```

Comment:
let's make this private final and make the format `HeadLimiter-pool-%d`

```suggestion
    private final ThreadFactory threadFactory = new ThreadFactoryBuilder().setNameFormat("HeadLimiter-pool-%d").build();
```